### PR TITLE
Add settings for completion

### PR DESCRIPTION
**Pull request by zappathustra**
Create settings to control completion:

- `completion.history_order` to sort items according to the database columns: visits, last_visit, title, uri (not sure the last two are interesting, but they don't hurt).
- `completion.max_items` to limit the number of items displayed.

More could be done, but that's what I needed at least.

### CODE REVIEW

File: lib/completion.lua [hunk 1]
```diff
@@ -290,14 +290,42 @@ local function sql_like_globber(term)
...
+    },
+    ["completion.max_items"] = {
+        type = "number", min = 1,
+        default = 25,
+        desc = "Number of completion items for history and bookmarks."
+    }
+})
+
+
+
```
Comment (McSinyx) [inline lib/completion.lua]: Man, why 2 trailing newlines:question:

Comment (zappathustra) [reply lib/completion.lua]: Bad me :)
Cleaned up.

File: lib/completion.lua [hunk 2]
```diff
@@ -290,14 +290,40 @@ local function sql_like_globber(term)
     return "%" .. escaped:gsub("%s+", "%%") .. "%"
 end
 
+settings.register_settings({
+    ["completion.history_order"] = {
```
Comment (aidanholm) [inline lib/completion.lua]: I'd prefer `completion.history.order` here

File: lib/completion.lua [hunk 3]
```diff
@@ -290,14 +290,40 @@ local function sql_like_globber(term)
...
+        type = "string",
+        default = "visits",
+        desc = [=[
+            A string indicating how history items should be sorted in
+            completion. Possible values are:<br>
```
Comment (aidanholm) [inline lib/completion.lua]: - desc uses markdown, so you shouldn't need to use `<br>` tags
- the keyswords (`visits`, `last_visit`, etc.) should be wrapped in backticks so they have code formatting

File: lib/completion.lua [hunk 4]
```diff
@@ -290,14 +290,40 @@ local function sql_like_globber(term)
...
+            - visits:     most visited URI's first<br>
+            - last_visit: most recent URI's first<br>
+            - title:      alphabetical sort<br>
+            - uri:        alphabetical sort
+        ]=],
+        validator = function (v)
+            local t = {visits = true, last_visit = true, title = true, uri = true}
```
Comment (aidanholm) [inline lib/completion.lua]: :+1:

**Review by aidanholm — COMMENTED** [review]
This looks pretty great! Just some formatting nits

**Comment by aidanholm** [comment]
Thanks! Merged this one manually

**Pull request closed without merging**